src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -0,0 +1,258 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "base/Base.h"
+#include "base/Status.h"
+#include "graph/UpdateEdgeExecutor.h"
+#include "meta/SchemaManager.h"
+#include "dataman/RowReader.h"
+#include "dataman/RowWriter.h"
+#include "storage/client/StorageClient.h"
+
+namespace nebula {
+namespace graph {
+
+// UPDATE/UPSERT EDGE <vertex_id> -> <vertex_id> [@<ranking>] OF <edge_type>
+// SET <update_decl> [WHEN <conditions>] [YIELD <field_list>]
+UpdateEdgeExecutor::UpdateEdgeExecutor(Sentence *sentence,
+                                       ExecutionContext *ectx) : Executor(ectx) {
+    sentence_ = static_cast<UpdateEdgeSentence*>(sentence);
+}
+
+
+Status UpdateEdgeExecutor::prepare() {
+    DCHECK(sentence_ != nullptr);
+    Status status = Status::OK();
+    do {
+        status = checkIfGraphSpaceChosen();
+        if (!status.ok()) {
+            break;
+        }
+        insertable_ = sentence_->getInsertable();
+        status = sentence_->getSrcId()->prepare();
+        if (!status.ok()) {
+            break;
+        }
+        auto src = sentence_->getSrcId()->eval();
+        if (src.ok() && !Expression::isInt(src.value())) {
+            status = Status::Error("SRC Vertex ID should be of type integer");
+            break;
+        }
+        edge_.set_src(Expression::asInt(src.value()));
+
+        status = sentence_->getDstId()->prepare();
+        if (!status.ok()) {
+            break;
+        }
+        auto dst = sentence_->getDstId()->eval();
```

Comment:
what about dst.ok() is false?

Reply:
Please note the **dst.ok()** in the If-Clause. :)
```
 50         auto dst = sentence_->getDstId()->eval();
 51         if (dst.ok() && !Expression::isInt(dst.value())) {
 52             status = Status::Error("DST Vertex ID should be of type integer");
 53             break;
 54         }
```

Reply:
maybe is if (!dst.ok() || !Expression::isInt(dst.value()))

Reply:
thx. it's good.